src/components/Button/index.js

```diff
@@ -279,29 +284,29 @@ class Button extends Component {
                 onMouseDown={this.props.onMouseDown}
                 disabled={this.props.isLoading || this.props.isDisabled}
                 wrapperStyle={[
-                    this.props.isDisabled ? {...styles.cursorDisabled, ...styles.noSelect} : {},
-                    styles.buttonContainer,
-                    this.props.shouldRemoveRightBorderRadius ? styles.noRightBorderRadius : undefined,
-                    this.props.shouldRemoveLeftBorderRadius ? styles.noLeftBorderRadius : undefined,
+                    this.props.isDisabled ? {...this.props.themeStyles.cursorDisabled, ...this.props.themeStyles.noSelect} : {},
+                    this.props.themeStyles.buttonContainer,
+                    this.props.shouldRemoveRightBorderRadius ? this.props.themeStyles.noRightBorderRadius : undefined,
+                    this.props.shouldRemoveLeftBorderRadius ? this.props.themeStyles.noLeftBorderRadius : undefined,
                     ...StyleUtils.parseStyleAsArray(this.props.style),
                 ]}
                 style={[
-                    styles.button,
-                    this.props.small ? styles.buttonSmall : undefined,
-                    this.props.medium ? styles.buttonMedium : undefined,
-                    this.props.large ? styles.buttonLarge : undefined,
-                    this.props.success ? styles.buttonSuccess : undefined,
-                    this.props.danger ? styles.buttonDanger : undefined,
-                    this.props.isDisabled && (this.props.success || this.props.danger) ? styles.buttonOpacityDisabled : undefined,
-                    this.props.isDisabled && !this.props.danger && !this.props.success ? styles.buttonDisabled : undefined,
-                    this.props.shouldRemoveRightBorderRadius ? styles.noRightBorderRadius : undefined,
-                    this.props.shouldRemoveLeftBorderRadius ? styles.noLeftBorderRadius : undefined,
+                    this.props.themeStyles.button,
```

Comment:
all the `this.props.themeStyles` will be shortened after this component is migrated right? I don't think `this.props.themeStyles.buttonContainer` with 4 words is very readable. Is there a workaround for class components?

Reply:
Yes, in class components i'd suggest to have something like

```js
const styles = this.props.themeStyles
const theme = this.props.theme
```

at the beginning of the `render` function.

I'll reframe the guidelines to actually favor this pattern.